### PR TITLE
Collab UI page

### PR DESCRIPTION
**Pull request by samyipsh**
Create own request
image
image

View own request
image

### CODE REVIEW

**Review by Nafeij — APPROVED** [review]
LGTM

File: src/pages/collab/index.tsx [hunk 1]
```diff
@@ -2,744 +2,540 @@
...
+  const onSubmit: SubmitHandler<CreateMatchRequestData> = (data) => {
+    onUpdate(data);
+  };
+
+  return (
+    <form
+      className="relative flex flex-col text-start items-stretch space-y-4 md:space-y-6 border p-4 rounded-md"
+      onSubmit={(e) => void handleSubmit(onSubmit)(e)}
+    >
+      e
```
Comment (kflim) [inline src/pages/collab/index.tsx]: I think this is by accident? :p

File: src/server/api/routers/matchRequest.ts [hunk 2]
```diff
@@ -455,83 +377,25 @@ export const matchRequestRouter = createTRPCRouter({
...
+      const matchedRequests = await ctx.prismaPostgres.matchRequest.findMany({
+        where: { userId: { in: [userId1, userId2] } },
       });
-      if (!r2)
+      if (matchedRequests.length !== 2) {
         throw new TRPCError({
           code: "BAD_REQUEST",
-          message: "Failed to match requests",
+          message: "Match request for both users not found",
         });
```
Comment (kflim) [inline src/server/api/routers/matchRequest.ts:392]: I'm not sure if we want both users to have match requests in the lobby before one is allowed to accept the other's request

**Review by kflim — COMMENTED** [review]
Some comments

File: src/pages/collab/index.tsx (hunk 1 above)
Comment (samyipsh) [reply src/pages/collab/index.tsx]: right thanks

File: src/server/api/routers/matchRequest.ts (hunk 2 above)
Comment (samyipsh) [reply src/server/api/routers/matchRequest.ts:392]: oh i think ur right there is likely a race condition if we don't use queues here (if thats what you are saying)

Comment (samyipsh) [reply src/server/api/routers/matchRequest.ts:392]: thanks for brining it up, lets modify it in a separate PR cos this one is quite big already?

**Comment by samyipsh** [comment]
@kflim 
^ potential race condition but merged cos this branch is mainly UI changes
lets do the fix in another PR with the redirecting to code session page (@Gabau)

**Pull request merged**